smithy-swift-codegen/src/main/kotlin/software/amazon/smithy/swift/codegen/middleware/MiddlewareExecutionGenerator.kt

```diff
@@ -58,18 +58,18 @@ class MiddlewareExecutionGenerator(
             }
             """.trimIndent()
         )
-        // Swift can't infer the generic arguments to `create` for some reason
-        writer.write(
-            """
-            config.httpInterceptorProviders.forEach { provider in
-                let i: any ${'$'}N<${'$'}N, ${'$'}N> = provider.create()
-                builder.interceptors.add(i)
-            }
-            """.trimIndent(),
-            ClientRuntimeTypes.Core.HttpInterceptor,
-            inputShape,
-            outputShape,
-        )
+        writer.openBlock(
+            "config.httpInterceptorProviders.forEach { (provider: any \$N) -> Void in",
```

Comment:
Cleaned up the multiline `write()` call while adding the type info to the `forEach` body block.